p2p/starknet/handlers_test.go

```diff
@@ -0,0 +1,130 @@
+package starknet_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/NethermindEth/juno/adapters/core2p2p"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/mocks"
+	"github.com/NethermindEth/juno/p2p/starknet"
+	"github.com/NethermindEth/juno/p2p/starknet/spec"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+)
+
+func TestHandleGetEvents(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	log := utils.NewNopZapLogger()
+	mockReader := mocks.NewMockReader(mockCtrl)
+
+	h := starknet.NewHandler(mockReader, log)
+
+	t.Run("nil block id", func(t *testing.T) {
+		_, err := h.HandleGetEvents(&spec.GetEvents{})
+		assert.Error(t, err)
+	})
+	t.Run("block by hash", func(t *testing.T) {
+		hash := new(felt.Felt).SetUint64(77)
+		mockReader.EXPECT().BlockByHash(hash).Return(&core.Block{}, nil)
+
+		expected := &spec.Events{}
+
+		events, err := h.HandleGetEvents(&spec.GetEvents{
+			Id: &spec.BlockID{
+				Hash: core2p2p.AdaptHash(hash),
+			},
+		})
+		require.NoError(t, err)
+		assert.True(t, proto.Equal(expected, events))
+	})
+	t.Run("block by height", func(t *testing.T) {
+		t.Run("failure", func(t *testing.T) {
+			height := uint64(777)
+			someErr := errors.New("some error")
+			mockReader.EXPECT().BlockByNumber(height).Return(nil, someErr)
+
+			_, err := h.HandleGetEvents(&spec.GetEvents{
+				Id: &spec.BlockID{
+					Height: height,
+				},
+			})
+			assert.True(t, errors.Is(err, someErr))
```

Comment:
```suggestion
			assert.ErrorIs(t, err, someErr)
```
I might be mis-remembering the name, but I'm pretty sure there is a helper for this.

Reply:
Thanks, added.